src/components/item/sharing/SharingLink.tsx

```diff
@@ -115,46 +118,63 @@ const SharingLink = ({ itemId }: Props): JSX.Element => {
     setLinkType(event.target.value as Context);
   };
-
+  const closeQrModal = () => {
+    setOpenQrModal(false);
+  };
+  const generateQrCode = () => {
+    setOpenQrModal(true);
+  };
   return (
-    <StyledBox
-      direction="row"
-      justifyContent="space-between"
-      alignItems="center"
-    >
-      <StyledLink href={link} target="_blank" id={SHARE_ITEM_DIALOG_LINK_ID}>
-        {link}
-      </StyledLink>
-      <Stack direction="row" alignItems="center">
-        <Select
-          sx={{ ml: 1, textTransform: 'capitalize' }}
-          value={linkType}
-          onChange={handleLinkTypeChange}
-          renderValue={(value) => enumT(value)}
-          id={SHARE_ITEM_DIALOG_LINK_SELECT_ID}
-        >
-          <MenuItem
-            // sx={{ textTransform: 'capitalize' }}
-            value={Context.Builder}
+    <>
+      <StyledBox
+        direction="row"
+        justifyContent="space-between"
+        alignItems="center"
+      >
+        <StyledLink href={link} target="_blank" id={SHARE_ITEM_DIALOG_LINK_ID}>
+          {link}
+        </StyledLink>
+        <Stack direction="row" alignItems="center">
+          <Select
+            sx={{ ml: 1, textTransform: 'capitalize' }}
+            value={linkType}
+            onChange={handleLinkTypeChange}
+            renderValue={(value) => enumT(value)}
+            id={SHARE_ITEM_DIALOG_LINK_SELECT_ID}
           >
-            {enumT(Context.Builder)}
-          </MenuItem>
-          <MenuItem
-            // sx={{ textTransform: 'capitalize' }}
-            value={Context.Player}
+            <MenuItem
+              // sx={{ textTransform: 'capitalize' }}
+              value={Context.Builder}
+            >
+              {enumT(Context.Builder)}
+            </MenuItem>
+            <MenuItem
+              // sx={{ textTransform: 'capitalize' }}
+              value={Context.Player}
+            >
+              {enumT(Context.Player)}
+            </MenuItem>
+          </Select>
+          <Tooltip
+            title={translateBuilder(BUILDER.SHARE_ITEM_LINK_COPY_TOOLTIP)}
           >
-            {enumT(Context.Player)}
-          </MenuItem>
-        </Select>
-        <Tooltip title={translateBuilder(BUILDER.SHARE_ITEM_LINK_COPY_TOOLTIP)}>
-          <span>
-            <IconButton onClick={handleCopy}>
-              <FileCopyIcon />
+            <span>
+              <IconButton onClick={handleCopy}>
+                <FileCopyIcon />
+              </IconButton>
+            </span>
+          </Tooltip>
+          <Tooltip title={translateBuilder(BUILDER.SHARE_ITEM_LINK_QR_CODE)}>
+            <IconButton onClick={generateQrCode}>
+              <QrCode2 />
             </IconButton>
-          </span>
-        </Tooltip>
-      </Stack>
-    </StyledBox>
+          </Tooltip>
+        </Stack>
+      </StyledBox>
+      {link && openQrModal && (
+        <QRCode open={openQrModal} handleClose={closeQrModal} value={link} />
```

Comment:
Would it make sense to create a `QrCodeButton` component that would contain 
```
<IconButton onClick={generateQrCode}>
              <QrCode2 />
            </IconButton>
```
and the modal?